src/query/string_opfunc.c

```diff
@@ -25033,9 +25034,13 @@ db_get_like_optimization_bounds (const DB_VALUE * const pattern, DB_VALUE * boun
 	}
       else
 	{
+	  /* Looking for an upper bound, therefore there is an LT compare.  */
+	  db_value_set_compare (bound, 1);
```

Comment:
I have a few comments on the db_value_set_compare function. I think we do not need it. I think we can modify the max string flag (currently compare LT) in db_value_domain_max directly:

      /* TODO: The string "\377" (one character of code 255) is not a perfect representation of the maximum value of a
       * string's domain. We should find a better way to do this. */
    case DB_TYPE_BIT:
    case DB_TYPE_VARBIT:
      value->data.ch.info.style = MEDIUM_STRING;
      value->data.ch.info.codeset = INTL_CODESET_RAW_BITS;
      value->data.ch.medium.size = 1;
      value->data.ch.medium.buf = (char *) "\377";	/* 255 */
      value->domain.general_info.is_null = 0;
      break;
      /* case DB_TYPE_STRING: internally DB_TYPE_VARCHAR */
    case DB_TYPE_CHAR:
    case DB_TYPE_VARCHAR:
    case DB_TYPE_NCHAR:
    case DB_TYPE_VARNCHAR:
      value->data.ch.info.style = MEDIUM_STRING;
      value->data.ch.info.codeset = codeset;
      if (codeset == INTL_CODESET_UTF8)
	{
	  value->data.ch.medium.size = 4;
	  /* maximum supported codepoint : check with intl_is_max_bound_chr */
	  value->data.ch.medium.buf = (char *) "\xf4\x8f\xbf\xbf";
	}
      else if (codeset == INTL_CODESET_KSC5601_EUC)
	{
	  value->data.ch.medium.size = 2;
	  value->data.ch.medium.buf = (char *) "\377\377";	/* 255 255 */
	}
      else
	{
	  value->data.ch.medium.size = 1;
	  value->data.ch.medium.buf = (char *) "\377";	/* 255 */
	}
      value->domain.general_info.is_null = 0;
      value->domain.char_info.collation_id = collation_id;
      break;
=>
      /* TODO: The string "\377" (one character of code 255) is not a perfect representation of the maximum value of a
       * string's domain. We should find a better way to do this. */
    case DB_TYPE_BIT:
    case DB_TYPE_VARBIT:
      value->data.ch.info.style = MEDIUM_STRING;
      value->data.ch.info.codeset = INTL_CODESET_RAW_BITS;
      value->data.ch.info.compare_LT = true;
      value->domain.general_info.is_null = 0;
      break;
      /* case DB_TYPE_STRING: internally DB_TYPE_VARCHAR */
    case DB_TYPE_CHAR:
    case DB_TYPE_VARCHAR:
    case DB_TYPE_NCHAR:
    case DB_TYPE_VARNCHAR:
      value->data.ch.info.style = MEDIUM_STRING;
      value->data.ch.info.codeset = codeset;
      value->data.ch.info.compare_LT = true;
      value->domain.general_info.is_null = 0;
      value->domain.char_info.collation_id = collation_id;
      break;